drivers/hmis/app/models/hmis/hud/enrollment.rb

```diff
@@ -339,49 +339,51 @@ def data_collection_features
     end.compact
   end
 
+  # Occurrence Point Forms that are enabled for this Enrollment.
+  # Returns array of OpenStructs, which are resolved by the HmisSchema::OccurrencePointForm GQL type.
   def occurrence_point_forms
     # Get definitions for Occurrence Point forms, including inactive/retired (but excluding drafts)
     definitions = Hmis::Form::Definition.with_role(:OCCURRENCE_POINT).published_or_retired.latest_versions
-    # Get cdeds that this enrollment has CDE record(s) for. Do this in advance so we don't make extra trips to db
-    cdeds_this_enrollment_has = custom_data_element_definitions.pluck(:key).to_set
 
-    definitions.map do |definition|
-      # Choose the most specific instance for this enrollment
+    # Filter to only those Occurrence Point Forms that are enabled
+    structs = definitions.map do |definition|
+      # Choose the most specific Instance that enables this FormDefinition for this Enrollment
       best_instance = definition.instances.active.detect_best_instance_for_enrollment(enrollment: self)
-
-      # Check for legacy data. Skip the calculation if there is a current instance
-      has_legacy_data = best_instance ? false : legacy_occurrence_point_data?(definition, cdeds_this_enrollment_has)
-
-      next unless best_instance || has_legacy_data
+      next unless best_instance
 
       OpenStruct.new(
-        legacy: has_legacy_data && !best_instance,
+        legacy: false, # not legacy, because there is an active Form Instance enabling it
         definition: definition,
-        data_collected_about: best_instance&.data_collected_about || 'ALL_CLIENTS',
+        data_collected_about: best_instance.data_collected_about || 'ALL_CLIENTS',
       )
     end.compact
-  end
-
-  private def legacy_occurrence_point_data?(definition, cdeds_this_enrollment_has)
-    definition.walk_definition_nodes(as_open_struct: true) do |item|
-      next unless item.mapping.present?
 
-      record_type = item.mapping&.record_type
-      field_name = item.mapping&.field_name&.underscore
-      custom_field_key = item.mapping&.custom_field_key
-
-      next unless record_type == 'ENROLLMENT' || custom_field_key
```

Comment:
Perhaps we don't need to remove all of this logic, but can instead change this (old line 373) to `next unless custom_field_key` and remove lines 375-378? If I'm understanding correctly, occurrence point forms can collect custom data elements. And we've mostly taken the approach of having forms create their own CDEDs instead of sharing them across forms. So, we don't expect the bug impacting move-in date, path status, and date of engagement to impact CDEs.